server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/OperatorTreeGenerator.java

```diff
@@ -2237,31 +2284,134 @@ private List<Operator> dealWithConsumeAllChildrenPipelineBreaker(
                 MPP_DATA_EXCHANGE_MANAGER.createLocalSourceHandleForPipeline(
                     ((LocalSinkHandle) localSinkHandle).getSharedTsBlockQueue(),
                     context.getDriverContext()),
-                childSource.getPlanNodeId());
+                partialParentNode.getPlanNodeId());
         context
             .getTimeSliceAllocator()
             .recordExecutionWeight(sourceOperator.getOperatorContext(), 1);
-        children.add(sourceOperator);
+        parentPipelineChildren.add(sourceOperator);
+        afterwardsNodes.add(partialParentNode);
         context.addExchangeOperator(sourceOperator);
         finalExchangeNum += subContext.getExchangeSumNum() - context.getExchangeSumNum() + 1;
       }
+      // 3. deal with exchangeNode
+      for (int i = exchangeNodeIndex; i < node.getChildren().size(); i++) {
+        PlanNode exchangeNode = node.getChildren().get(i);
```

Comment:
Even if the previous assumption is right, I think it's better to do a safety-check here.
```
assert(exchangeNode instanceof ExchangeNode)
```